File: dotnet/RelevantCodes.ExtentReports/RelevantCodes.ExtentReports/Model/Media.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RelevantCodes.ExtentReports.Model
{
    internal class Media
    {
        public string TestName
        {
            get;
            internal set;
        }

        public string Source
        {
            get;
            internal set;
        }

        public Guid TestID
        {
            get;
            internal set;
        }

        public Media() { }

        public Media(string TestName, string Source, Guid TestID)
        {
            this.TestName = TestName;
            this.Source = Source;
            this.TestID = TestID;
        }
    }
}

File: src/main/java/com/relevantcodes/extentreports/model/Media.java
package com.relevantcodes.extentreports.model;

import java.io.Serializable;
import java.util.UUID;

public abstract class Media implements Serializable {
    
    private static final long serialVersionUID = 7703803003727779654L;

    public void setSource(String source) {
        this.source = source;
    }
    
    public String getSource() {
        return source;
    }
    
    public void setTestName(String testName) {
        this.testName = testName;
    }
    
    public String getTestName() {
        return testName;
    }
    
    public void setTestId(UUID id) {
        this.id = id;
    }
    
    public UUID getTestId() {
        return id;
    }
    
    private UUID id;
    private String source;
    private String testName;
}
